File: shifai-android/app/src/main/java/com/shifai/presentation/util/FrenchDate.kt
package com.shifai.presentation.util

import java.time.LocalDate
import java.time.LocalTime
import java.time.format.DateTimeFormatter
import java.time.temporal.ChronoUnit
import java.util.Locale

/**
 * French Date Formatters — centralized date formatting.
 * All formats use French locale.
 * Mirrors iOS FrenchDate.swift.
 */
object FrenchDate {

    private val frenchLocale = Locale.FRANCE

    // ─── Formatters ───

    /** "14 février 2026" */
    val full: DateTimeFormatter = DateTimeFormatter.ofPattern("d MMMM yyyy", frenchLocale)

    /** "14 fév. 2026" */
    val medium: DateTimeFormatter = DateTimeFormatter.ofPattern("d MMM yyyy", frenchLocale)

    /** "14/02/2026" */
    val short: DateTimeFormatter = DateTimeFormatter.ofPattern("dd/MM/yyyy", frenchLocale)

    /** "Lundi 14 février" */
    val dayAndMonth: DateTimeFormatter = DateTimeFormatter.ofPattern("EEEE d MMMM", frenchLocale)

    /** "Fév. 2026" */
    val monthYear: DateTimeFormatter = DateTimeFormatter.ofPattern("MMM yyyy", frenchLocale)

    /** "14:30" */
    val time: DateTimeFormatter = DateTimeFormatter.ofPattern("HH:mm", frenchLocale)

    // ─── Convenience ───

    fun cycleDay(day: Int, phase: String): String = "Jour $day — $phase"

    fun daysUntil(date: LocalDate): String {
        val days = ChronoUnit.DAYS.between(LocalDate.now(), date).toInt()
        return when {
            days == 0 -> "Aujourd'hui"
            days == 1 -> "Demain"
            days > 1 -> "Dans $days jours"
            days == -1 -> "Hier"
            else -> "Il y a ${-days} jours"
        }
    }

    fun formatFull(date: LocalDate): String = date.format(full)
    fun formatShort(date: LocalDate): String = date.format(short)
    fun formatTime(time: LocalTime): String = time.format(this.time)
}

File: shifai-ios/ShifAI/Presentation/Utilities/FrenchDate.swift
import Foundation

/// French Date Formatters — centralized date formatting for the app.
/// All formats use French locale.
enum FrenchDate {

    private static let frenchLocale = Locale(identifier: "fr-FR")

    // MARK: - Formatters

    /// "14 février 2026"
    static let full: DateFormatter = {
        let f = DateFormatter()
        f.locale = frenchLocale
        f.dateStyle = .long
        f.timeStyle = .none
        return f
    }()

    /// "14 fév. 2026"
    static let medium: DateFormatter = {
        let f = DateFormatter()
        f.locale = frenchLocale
        f.dateStyle = .medium
        f.timeStyle = .none
        return f
    }()

    /// "14/02/2026"
    static let short: DateFormatter = {
        let f = DateFormatter()
        f.locale = frenchLocale
        f.dateFormat = "dd/MM/yyyy"
        return f
    }()

    /// "Lundi 14 février"
    static let dayAndMonth: DateFormatter = {
        let f = DateFormatter()
        f.locale = frenchLocale
        f.dateFormat = "EEEE d MMMM"
        return f
    }()

    /// "Fév. 2026"
    static let monthYear: DateFormatter = {
        let f = DateFormatter()
        f.locale = frenchLocale
        f.dateFormat = "MMM yyyy"
        return f
    }()

    /// "14:30"
    static let time: DateFormatter = {
        let f = DateFormatter()
        f.locale = frenchLocale
        f.dateFormat = "HH:mm"
        return f
    }()

    // MARK: - Convenience

    static func relative(_ date: Date) -> String {
        let formatter = RelativeDateTimeFormatter()
        formatter.locale = frenchLocale
        formatter.unitsStyle = .full
        return formatter.localizedString(for: date, relativeTo: Date())
    }

    static func cycleDay(_ day: Int, phase: String) -> String {
        "Jour \(day) — \(phase)"
    }

    static func daysUntil(_ date: Date) -> String {
        let days = Calendar.current.dateComponents([.day], from: Date(), to: date).day ?? 0
        if days == 0 { return "Aujourd'hui" }
        if days == 1 { return "Demain" }
        return "Dans \(days) jours"
    }
}
